### PR TITLE
\Magento\Framework\Filesystem\Io\File::read() should be able to accept resource in second argument

### PR DESCRIPTION
**Pull request by bnymn**
\Magento\Framework\Filesystem\Io\File::read() method is not compatible with the `IoInterface`. This PR aims to make it compatible.

### Fixed Issues (if relevant)

1. magento/magento2#27866: \Magento\Framework\Filesystem\Io\File::read() method is not compatible with \Magento\Framework\Filesystem\Io\IoInterface::read()

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: lib/internal/Magento/Framework/Filesystem/Test/Unit/Io/FileTest.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        }
+        return $tmpDir;
+    }
+
+    /**
+     * To cover the issue on GitHub: #27866
+     * @test
+     * @throws LocalizedException
+     */
+    public function read_should_copy_the_source_file_to_the_given_file_resource()
```
Comment (lenaorobei) [inline lib/internal/Magento/Framework/Filesystem/Test/Unit/Io/FileTest.php]: Please follow `camelCase` naming.

File: lib/internal/Magento/Framework/Filesystem/Test/Unit/Io/FileTest.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
+<?php
+namespace Magento\Framework\Filesystem\Test\Unit\Io;
+
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\Filesystem\Io\File;
+use PHPUnit\Framework\TestCase;
+
+class FileTest extends TestCase
+{
+    protected function setUp()
```
Comment (lenaorobei) [inline lib/internal/Magento/Framework/Filesystem/Test/Unit/Io/FileTest.php]: Looks like redundant.

**Review by lenaorobei — CHANGES_REQUESTED** [review]
Please sign CLA and address test failures.

**Comment by engcom-Charlie** [comment]
Hi @bnymn, please sign CLA and look through failed tests. Otherwise, we can't proceed with your PR.

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Comment by lenaorobei** [comment]
@magento run all tests

**Comment by lenaorobei** [comment]
@magento run Static Tests

**Comment by engcom-Charlie** [comment]
Hi, @bnymn can you please fix static tests?
Thank you.

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Review by lenaorobei — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @lenaorobei, thank you for the review.
[ENGCOM-8127](https://jira.corp.magento.com/browse/ENGCOM-8127) has been created to process this Pull Request

**Comment by engcom-Delta** [comment]
:heavy_check_mark: QA passed
**Result:**
Code snippet for testing:
```
$file = ObjectManager::getInstance()->get(File::class);
        if (!is_dir('var/tmp_1')) {
            // dir doesn't exist, make it
            mkdir('var/tmp_1');
        }
        file_put_contents('var/tmp_1/sample_file', 'Some test file');
        $targetHandle = fopen('var/tmp_1/' . rand(), 'w');
        $file->cd('var/tmp_1');
        $result = $file->read('sample_file', $targetHandle);
        $output->writeln($result);

```
**Before:**
:x: $result is `null`
image

**After:**
:heavy_check_mark: File content is copied to the target path
image

:heavy_check_mark: File is copied to the target file when second argument is string only: `$result = $file->read('sample_file', 'test');
image

**Comment by engcom-Delta** [comment]
**Note:** Automation tests are passed

**Pull request merged**